### PR TITLE
Merge pull request #5 from Rashdan98/chapter_6

### PR DESCRIPTION
**Pull request by Rashdan98**
Merge pull request #6  from Rashdan98/chapter_6

### CODE REVIEW

**Comment by Rashdan98** [comment]
ready

**Review by ghost — COMMENTED** [review]
You need to create new commit for each branch,
you are using the merged commit from previous branch
Please check

**Comment by Rashdan98** [comment]
ready

**Review by ghost — COMMENTED** [review]
you probably missing some file in chapter 6

**Comment by Rashdan98** [comment]
> you probably missing some file in chapter 6

is it file schema.rb?

**Comment by Rashdan98** [comment]
ready

**Review by ghost — COMMENTED** [review]
> > you probably missing some file in chapter 6
> 
> is it file schema.rb?
You don't have user.rb and some file migrate

**Comment by Rashdan98** [comment]
ready

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class User < ApplicationRecord
+    before_save {self.email = email.downcase}
```
Comment (huongnt-2545) [inline app/models/user.rb]: wrong indent from line 2 -> 11, recommend use 2 spaces instead of 4 spaces

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+    
+    validates :name, presence: true
+    VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+    validates :email, presence: true, length: {maximum: 255}, format: {with: VALID_EMAIL_REGEX}
+    validates :email, uniqueness: true
+    
+    has_secure_password
+    validates :password, presence: true, length: { minimum: 6 }
+  end
+
```
Comment (huongnt-2545) [inline app/models/user.rb]: remove all redundant spaces in this line

File: app/views/layouts/application.html.erb [hunk 3]
```diff
@@ -1,20 +1,19 @@
...
-    <div class="container">
-      <%= yield %>
-    </div>
-
+  <div class="container">
+   <%= yield %>
+   </div>
   </body>
-</html>
+</html>
```
Comment (huongnt-2545) [inline app/views/layouts/application.html.erb]: add empty line at the end of file

**Comment by Rashdan98** [comment]
ready

File: db/202110080182816_add_password_digest_to_users.rb [hunk 4]
```diff
@@ -0,0 +1,6 @@
+class AddPasswordDigestToUsers < ActiveRecord::Migration[6.1]
+    def change
+      add_column :users, :password_digest, :string
+    end
+  end
+
```
Comment (ghost) [inline db/202110080182816_add_password_digest_to_users.rb]: leave the last line blank

File: app/views/layouts/application.html.erb [hunk 5]
```diff
@@ -1,20 +1,19 @@
...
 
   <body>
     <%= render "layouts/header" %>
-    <div class="container">
-      <%= yield %>
-    </div>
-
+  <div class="container">
+   <%= yield %>
+   </div>
```
Comment (ghost) [inline app/views/layouts/application.html.erb]: indent please

**Review by ghost — COMMENTED** [review]
cmt

**Comment by Rashdan98** [comment]
ready

**Review by huongnt-2545 — APPROVED** [review]
merged
You can merge this pull

**Pull request merged**